scipy/special/_gufuncs.cpp

```diff
@@ -0,0 +1,105 @@
+extern const char *lqmn_doc;
+extern const char *rctj_doc;
+extern const char *rcty_doc;
+
+// This is needed by sf_error, it is defined in the Cython "_ufuncs_extra_code_common.pxi" for "_generate_pyx.py".
+// It exists to "call PyUFunc_getfperr in a context where PyUFunc_API array is initialized", but here we are
+// already in such a context.
+extern "C" int wrap_PyUFunc_getfperr() { return PyUFunc_getfperr(); }
+
+static PyModuleDef _gufuncs_def = {
+    PyModuleDef_HEAD_INIT,
+    .m_name = "_gufuncs",
+    .m_size = -1,
+};
+
+PyMODINIT_FUNC PyInit__gufuncs() {
+    if (!SpecFun_Initialize()) {
+        return nullptr;
+    }
+
+    PyObject *_gufuncs = PyModule_Create(&_gufuncs_def);
+    if (_gufuncs == nullptr) {
+        return nullptr;
+    }
+
+    PyObject *_lpn =
+        SpecFun_NewGUFunc({static_cast<func_f_f1f1_t>(special::lpn), static_cast<func_d_d1d1_t>(special::lpn),
+                           static_cast<func_F_F1F1_t>(special::lpn), static_cast<func_D_D1D1_t>(special::lpn)},
+                          2, "_lpn", lpn_doc, "()->(np1),(np1)");
+    PyModule_AddObjectRef(_gufuncs, "_lpn", _lpn);
+
+    PyObject *_lpmn =
+        SpecFun_NewGUFunc({static_cast<func_fl_f2f2_t>(special::lpmn), static_cast<func_dl_d2d2_t>(special::lpmn)}, 2,
+                          "_lpmn", lpmn_doc, "(),()->(mp1,np1),(mp1,np1)");
+    PyModule_AddObjectRef(_gufuncs, "_lpmn", _lpmn);
+
+    PyObject *_clpmn =
+        SpecFun_NewGUFunc({static_cast<func_Fll_F2F2_t>(special::clpmn), static_cast<func_Dll_D2D2_t>(special::clpmn)},
+                          2, "_clpmn", clpmn_doc, "(),(),()->(mp1,np1),(mp1,np1)");
+    PyModule_AddObjectRef(_gufuncs, "_clpmn", _clpmn);
+
+    PyObject *_lqn =
+        SpecFun_NewGUFunc({static_cast<func_f_f1f1_t>(special::lqn), static_cast<func_d_d1d1_t>(special::lqn),
+                           static_cast<func_F_F1F1_t>(special::lqn), static_cast<func_D_D1D1_t>(special::lqn)},
+                          2, "_lqn", lqn_doc, "()->(np1),(np1)");
+    PyModule_AddObjectRef(_gufuncs, "_lqn", _lqn);
+
+    PyObject *_lqmn =
+        SpecFun_NewGUFunc({static_cast<func_f_f2f2_t>(special::lqmn), static_cast<func_d_d2d2_t>(special::lqmn),
+                           static_cast<func_F_F2F2_t>(special::lqmn), static_cast<func_D_D2D2_t>(special::lqmn)},
+                          2, "_lqmn", lqmn_doc, "()->(mp1,np1),(mp1,np1)");
+    PyModule_AddObjectRef(_gufuncs, "_lqmn", _lqmn);
+
+    PyObject *_rctj =
+        SpecFun_NewGUFunc({static_cast<func_f_f1f1_t>(special::rctj), static_cast<func_d_d1d1_t>(special::rctj)}, 2,
+                          "_rctj", rctj_doc, "()->(np1),(np1)");
+    PyModule_AddObjectRef(_gufuncs, "_rctj", _rctj);
+
+    PyObject *_rcty =
+        SpecFun_NewGUFunc({static_cast<func_f_f1f1_t>(special::rcty), static_cast<func_d_d1d1_t>(special::rcty)}, 2,
+                          "_rcty", rcty_doc, "()->(np1),(np1)");
+    PyModule_AddObjectRef(_gufuncs, "_rcty", _rcty);
```

Comment:
So while I've added the Ricatti-Bessel functions, I actually haven't changed their external behaviour at all. Meaning they still only take scalars and error otherwise. This is just taking them from Cython bindings to gufunc binding. In a future PR, we'll open them up to arrays.